examples/apps/generic_client/generic_client.py

```diff
@@ -119,16 +121,18 @@ def ParseCommandLine(args) :
 			sys.exit(-1)
 		else:
 			logger.error("Mode should be either registry or listing")
+			sys.exit(-1)
 
 	else:
 		if not mode:
 			logger.error("Address needs to be passed with mode option")
+			sys.exit(-1)
```

Comment:
Will the mode be set to default "registry" if not explicitly set with the above changes. If so, mode can never be None. Hence "if not mode" condition check is not required I think.

Reply:
I removed the code (lines 122-129 above, both `else` clauses)

Reply:
Correct.  I removed the `if not mode` check.